### ISSUE TITLE
 Assessment of the difficulty in porting CPU architecture for kokkos

### ISSUE THREAD

**Issue by wangyuliu** [kokkos/kokkos#6414]
Hello everyone! I am working on implementing a tool to assess the complexity of CPU architecture porting. It primarily focuses on RISC-V architecture porting.In fact, it can be estimated on average across various architectures. As part of my dataset, I have collected the **kokkos** project. **I would like to gather community opinions to support my assessment. I appreciate your help and response!** Based on scanning tools, the porting complexity is determined to be simple, with a very small amount of code related to the CPU architecture in the project. (referring to the overall workload from adapting the project to a specific architecture to achieving full functionality on that architecture). Is this assessment accurate? I look forward to your help and response.

**Comment by crtrott** [comment]
Yes, we don't expect any issues since there is really not much if any assembly which is required to b provided. The only thing would be RISC specific SIMD types. A number of folks already played around with RISK and Kokkos btw: see #6323 for example.

**Comment by diehlpk** [comment]
@wangyuliu We played with kokkos and RISC-V, see [here](https://arxiv.org/abs/2309.06530). We have one section about "porting" kokkos to RISC-V  development boards. I am happy to discuss with you.
